src/color.rs

```diff
@@ -55,6 +45,33 @@ pub fn serialize_color_alpha(
     rounded_alpha.to_css(dest)
 }
 
+/// A [`ModernComponent`] can serialize to `none`, `nan`, `infinity` and
+/// floating point values.
+struct ModernComponent<'a>(&'a Option<f32>);
+
+impl<'a> ToCss for ModernComponent<'a> {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result
+    where
+        W: fmt::Write,
+    {
+        if let Some(value) = self.0 {
+            if value.is_infinite() {
+                if value.is_sign_negative() {
+                    dest.write_str("calc(-infinity)")
```

Comment:
Hmm, so the behavior of out-of-range negative values like `-3` would be different than `-infinity`? Or would it be consistent?

Reply:
`-3` would serialize to `-3` and `-infinity` would serialize to `calc(-infinity)`.

As far as the parser is concerned now, there is no range, just numbers between -inf..inf.  Up to the user of the lib to clamp/decide what is a range.

Reply:
Ok, we should make sure that Gecko is also consistent in how we treat these but that seems fine.